backend/server.js

```diff
@@ -0,0 +1,322 @@
+app.use(cors());
+app.use(express.json());
+
+// Our own middleware that checks if the database is connected before going forward to our endpoints
+app.use((req, res, next) => {
+  if (mongoose.connection.readyState === 1) {
+    next();
+  } else {
+    res.status(503).json({ error: "Service unavailable" });
+  }
+});
+
+// Add models for the db
+const UserSchema = new mongoose.Schema({
+  username: {
+    type: String,
+    unique: true,
+    required: true,
+  },
+  password: {
+    type: String,
+    required: true,
+  },
+  accessToken: {
+    type: String,
+    default: () => crypto.randomBytes(128).toString("hex"),
+  },
+});
+
+const RecipeSchema = new mongoose.Schema({
+  name: {
+    type: String,
+    unique: true,
+    required: true,
+  },
+  description: String,
+  image: String,
+  recipeCategory: String,
+  recipeCookingTime: String,
+  recipeMainIngredient: String,
+  recipeIngredients: [
+    {
+      recipeIngredientAmount: String,
+      recipeIngredientUnit: String,
+      recipeIngredient: String,
+    },
+  ],
+  recipeInstruction: [
+    {
+      instruction: Array,
+    },
+  ],
+  datePublished: {
+    type: Date,
+    default: Date.now(),
+  },
+  source: String,
+  likes: {
+    type: Number,
+    default: 0,
+  },
+});
+
+const User = mongoose.model("User", UserSchema);
+const Recipe = mongoose.model("Recipe", RecipeSchema);
+
+// authenticates user with access token
+const authenticateUser = async (req, res, next) => {
+  const accessToken = req.header("Authorization");
+
+  try {
+    const user = await User.findOne({ accessToken });
+    if (user) {
+      next();
+    } else {
+      res.status(401).json({ response: "Vänligen logga in", success: false });
+    }
+  } catch (error) {
+    res.status(400).json({ response: error, success: false });
+  }
+};
+
+// Start defining your routes here
+app.get("/", (req, res) => {
+  res.send({
+    "Welcome to Molbimiens skafferi API - by Maria Sjögren (https://github.com/molbimien/). See full documentation here 👉 https://github.com/molbimien/final-project-molbimien-recipes/blob/main/Documentation.md":
+      listEndpoints(app),
+  });
+});
+
+// endpoint to get all recipes
+app.get("/recipes", async (req, res) => {
+  const recipe = await Recipe.find(req.query);
+  res.json(recipe);
+});
+
+// endpoint to get the latest recipes
+app.get("/recipes/latest", async (req, res) => {
+  const latest = await Recipe.find(req.params.datePublished)
+    .sort({ datePublished: -1 }) // -1 for descending sort
+    .limit(3);
+
+  res.json(latest);
+});
+
+// endpoint to get the most liked recipes
+app.get("/recipes/liked", async (req, res) => {
+  const likes = await Recipe.find(req.params.likes)
+    .sort({ likes: -1 }) // -1 for descending sort
+    .limit(3);
+
+  res.json(likes);
+});
+
+// endpoint to get one recipe based on id
+app.get("/recipes/id/:id", async (req, res) => {
+  try {
+    const recipeById = await Recipe.findById(req.params.id);
```

Comment:
It would be great to destructure properties from `req` object in those endpoints!